pkg/app/test_command.go

```diff
@@ -6,130 +6,136 @@ import (
 	"log"
 	"os"
 	"path"
-	"sync"
 
 	"github.com/SimonBaeumer/commander/pkg/output"
 	"github.com/SimonBaeumer/commander/pkg/runtime"
 	"github.com/SimonBaeumer/commander/pkg/suite"
 )
 
+var out output.OutputWriter
+
 // TestCommand executes the test argument
 // testPath is the path to the test suite config, it can be a dir or file
 // titleFilterTitle is the title of test which should be executed, if empty it will execute all tests
 // ctx holds the command flags. If directory scanning is enabled with --dir it is
 // not supported to filter tests, therefore testFilterTitle is an empty string
 func TestCommand(testPath string, testFilterTitle string, ctx AddCommandContext) error {
-	if ctx.Verbose == true {
+	if ctx.Verbose {
 		log.SetOutput(os.Stdout)
 	}
 
+	out = output.NewCliOutput(!ctx.NoColor)
+
 	if testPath == "" {
 		testPath = CommanderFile
 	}
 
-	var results <-chan runtime.TestResult
+	var result runtime.Result
 	var err error
-	if ctx.Dir {
+	switch {
+	case ctx.Dir:
 		if testFilterTitle != "" {
 			return fmt.Errorf("Test may not be filtered when --dir is enabled")
 		}
 		fmt.Println("Starting test against directory: " + testPath + "...")
 		fmt.Println("")
-		results, err = testDir(testPath)
-	} else {
+		result, err = testDir(testPath)
+	default:
 		fmt.Println("Starting test file " + testPath + "...")
 		fmt.Println("")
-		results, err = testFile(testPath, testFilterTitle)
+		result, err = testFile(testPath, "", testFilterTitle)
 	}
 
 	if err != nil {
 		return fmt.Errorf(err.Error())
 	}
 
-	out := output.NewCliOutput(!ctx.NoColor)
-	if !out.Start(results) {
+	if !out.PrintSummary(result) && !ctx.Verbose {
 		return fmt.Errorf("Test suite failed, use --verbose for more detailed output")
 	}
 
 	return nil
 }
 
-func testDir(directory string) (<-chan runtime.TestResult, error) {
+func testDir(directory string) (runtime.Result, error) {
+	result := runtime.Result{}
+
+	// TODO: validate entire dir before running tests: issue #129
```

Comment:
Removing TODO, it doen't add anything to the source because an issue exists. 
With these type of TODOs you create two sources of information, i.e. the issue gets closed the next programmer is confused by this comment.
The same reason applies of the `skip dirs` TODO comment.